app/connectors/GuaranteeBalanceConnector.scala

```diff
@@ -99,21 +97,43 @@ class GuaranteeBalanceConnector @Inject() (http: HttpClient, appConfig: Frontend
     )
   }
 
+  private def processSubmitErrorResponse(response: HttpResponse): Either[HttpResponse, BalanceRequestResponse] = {
+    val json = response.validateJson[PostBalanceRequestFunctionalErrorResponse]
+    val balanceRequestResponse = json match {
+      case JsSuccess(fe, _) => convertErrorTypeToBalanceRequestResponse(fe.response.errors)
+      case _                => None
+    }
+
+    balanceRequestResponse match {
+      case Some(brResponse) => Right(brResponse)
+      case None =>
+        logger.info(s"[GuaranteeBalanceConnector][processSubmitErrorResponse] ${json.fold(
+          _.toString(),
+          fe => s"Response contains functional error type(s) ${fe.errorTypes}"
+        )}")
+        Left(response)
+    }
+  }
+
   private def processQuerySuccessResponse(balanceId: BalanceId, response: HttpResponse): BalanceRequestResponse = {
     val balanceRequestResponse = response.json.as[GetBalanceRequestResponse].request
     balanceRequestResponse.response match {
       case Some(response) =>
         response match {
-          case fe: BalanceRequestFunctionalError if fe.containsErrorType(NotMatchedErrorType) =>
-            BalanceRequestNotMatched
-          case fe: BalanceRequestFunctionalError if fe.containsErrorType(UnsupportedGuaranteeTypeErrorType) =>
-            BalanceRequestUnsupportedGuaranteeType
-          case _ => response
+          case fe: BalanceRequestFunctionalError => convertErrorTypeToBalanceRequestResponse(fe.errors).getOrElse(response)
+          case _                                 => response
         }
       case _ =>
         val currentTime        = Instant.now()
         val timeRequestExpires = balanceRequestResponse.requestedAt.plusSeconds(appConfig.guaranteeBalanceExpiryTime)
         if (currentTime.isBefore(timeRequestExpires)) BalanceRequestPending(balanceId) else BalanceRequestPendingExpired(balanceId)
     }
   }
+
+  private def convertErrorTypeToBalanceRequestResponse(errorTypes: NonEmptyList[FunctionalError]): Option[BalanceRequestResponse] = {
+    val errorValues: List[BalanceRequestResponse] = errorTypes.toList.flatMap(
+      errorType => errorResponseMap.get(errorType.errorType)
+    )
+    errorValues.headOption
```

Comment:
Similarly, we could probably do without the val here

Reply:
Done